internal/build/buildconfig/manager.go

```diff
@@ -42,18 +50,27 @@ func (bcm *buildConfigManager) Sync(ctx context.Context, mod kmmv1beta1.Module,
 
 		logger.Info("Creating BuildConfig")
 
-		buildConfig, err = bcm.maker.MakeBuildConfig(mod, m, targetKernel, m.ContainerImage, pushImage)
-		if err != nil {
-			return build.Result{}, fmt.Errorf("could not make BuildConfig: %v", err)
-		}
-
-		if err = bcm.client.Create(ctx, buildConfig); err != nil {
-			return build.Result{}, fmt.Errorf("could not create Job: %v", err)
+		if err = bcm.client.Create(ctx, buildConfigTemplate); err != nil {
+			return build.Result{}, fmt.Errorf("could not create BuildConfig: %v", err)
 		}
 
 		return build.Result{Status: build.StatusCreated, Requeue: true}, nil
 	}
 
+	changed, err := bcm.isBuildConfigChanged(buildConfig, buildConfigTemplate)
+	if err != nil {
+		return build.Result{}, fmt.Errorf("could not determine if BuildConfig has changed: %v", err)
+	}
+
+	if changed {
+		logger.Info("The module's build spec has been changed, deleting the current BuildConfig so a new one can be created", "name", buildConfig.Name)
+		err = bcm.client.Delete(ctx, buildConfig)
```

Comment:
Don't we only need to delete the build and let the buildConfig generate a new one?

Reply:
i don't see any significant advantage in deleting the Build, and not BuildConfig:
1) deleting the BuildConfig makes the code simplier, since we already have that information, and don't need to to look for relevant Build
2) if we delete Build, then we need to edit BuildConfig first. This means additional code. In case we delete BuildConfig, no additional code for editing needed, since the code will simply create new BuildConfig in the next reconciliation loop

Reply:
Ack.